src/Domain/FunctionsOfTime/ThreadsafeList.tpp

```diff
@@ -0,0 +1,263 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "Domain/FunctionsOfTime/ThreadsafeList.hpp"
+
+#include <atomic>
+#include <memory>
+#include <pup.h>
+#include <pup_stl.h>
+#include <utility>
+
+#include "Utilities/EqualWithinRoundoff.hpp"
+#include "Utilities/ErrorHandling/Error.hpp"
+
+namespace domain::FunctionsOfTime::FunctionOfTimeHelpers {
+namespace ThreadsafeList_detail {
+template <typename T>
+struct Interval {
+  double expiration;
+  T data;
+  std::unique_ptr<Interval> previous;
+
+  void pup(PUP::er& p);
+};
+
+template <typename T>
+void Interval<T>::pup(PUP::er& p) {
+  p | expiration;
+  p | data;
+  p | previous;
+}
+}  // namespace ThreadsafeList_detail
+
+template <typename T>
+ThreadsafeList<T>::ThreadsafeList() = default;
```

Comment:
Thoughts on explicit instantiation in a `cpp` for this if we only plan to use it in the functions of time? I know for generic containers that's not the best because then you have to have an instantiation for each possible type you'll need. But if we only plan to use these in the functions of time, I could see it working for this (even though you're right, this wouldn't take up much compile time in an `hpp/tpp`).